src/socket.c

```diff
@@ -195,6 +195,14 @@ static int raw_connect_internal(struct trilogy_sock *sock, const struct addrinfo
         return TRILOGY_SYSERR;
     }
 
+#ifdef TCP_NODELAY
+    if (sock->base.opts.tcp_nodelay_enabled) {
+      int flags = 1;
+      if (setsockopt(sock->fd, IPPROTO_TCP, TCP_NODELAY, (void *)&flags, sizeof(flags)) < 0) {
+        goto fail;
+      }
+    }
+#endif
```

Comment:
The indentation is off for C code, should be indented one more time.

```suggestion
#ifdef TCP_NODELAY
    if (sock->base.opts.tcp_nodelay_enabled) {
        int flags = 1;
        if (setsockopt(sock->fd, IPPROTO_TCP, TCP_NODELAY, (void *)&flags, sizeof(flags)) < 0) {
            goto fail;
        }
    }
#endif
```